drivers/tee/optee/call.c

```diff
@@ -220,6 +220,16 @@ int optee_open_session(struct tee_context *ctx,
 	struct optee_msg_arg *msg_arg;
 	phys_addr_t msg_parg;
 	struct optee_session *sess = NULL;
+	uuid_t client_uuid;
+
+	rc = tee_session_verify_perm(arg->clnt_login, arg->clnt_uuid);
```

Comment:
Since this is how these two functions are supposed to be used, how about having only one that does it all?
Perhaps we could put the result directly at `&msg_arg->params[1].u.value` in order to save some stack?

Reply:
Of course this is now about OP-TEE path here but there is also a AMD path in kernel so I am wondering if that would need the same treatment and in there it might be different.

Basic idea was to keep policy and generation as a separate.

Reply:
Right now a single function would be best, so let's have that.
Please move the OP-TEE changes into a separate patch.

Reply:
ok